Lib/hmac.py

```diff
@@ -55,15 +52,32 @@ def __init__(self, key, msg=None, digestmod=''):
         if not digestmod:
             raise TypeError("Missing required parameter 'digestmod'.")
 
+        if (
+            _hashopenssl is not None and
+            (digestname := _hashlib._digestmod_to_name(digestmod))
+        ):
+            self._init_hmac(key, msg, digestname)
```

Comment:
This makes the assumption that if we have `_hashopenssl`, all possible digests anyone could use will be available as part of openssl for use via its hmac implementation.  I don't think that is necessarily true.  A digestmod that supplies a string name can be supplied but not be something available in openssl.

presumably the easiest way around this while retaining the optimal openssl computation when possible is to catch an exception from `_init_hmac` (due to `_hashopenssl.hmac_new` raising) and fall back to `_init_old` instead.

a test should be added to cover this case.  (via a custom digestmod-like-object with a name and blocksize perhaps)